plugins/module_utils/entity.py

```diff
@@ -372,6 +375,12 @@ def _upload_file(
             resp_json = None
 
         if not raise_error:
+            # Add error details and status details if any
+            if not resp_json:
+                resp_json = {}
+            if status_code >= 300:
+                resp_json["error"] = body
+            resp_json["status_code"] = status_code
```

Comment:
@premkarat Incase the image upload fails due to any reason, we need to delete the PC image created before. Here incase of error, _upload_file() stops the execution and throws error. With "raise_error" as False, we get response back but its None from api itself. So wanted to add this info in return so that module can do the cleanup if fails.